File: repository/src/main/java/com/sreMake/repository/system/CasbinRuleRepository.java
package com.sreMake.repository.system;

import com.sreMake.model.system.CasbinRule;
import org.babyfish.jimmer.spring.repo.JavaRepository;

import java.util.List;

public interface CasbinRuleRepository extends JavaRepository<CasbinRule, Integer> {
    /**
     * 查询casbin中有多少匹配到的
     */
    long countMatches(String router);

    /**
     * 根据角色查询资源
     */
    List<String> listMatchesRole(String... roles);
}

File: repository/src/main/kotlin/com/SReMake/repository/system/CasbinRuleRepository.kt
package com.SReMake.repository.system

import com.SReMake.model.system.CasbinRule
import org.babyfish.jimmer.spring.repo.KotlinRepository

interface CasbinRuleRepository : KotlinRepository<CasbinRule, Int> {
    /**
     * 查询casbin中有多少匹配到的
     */
    fun countMatches(router: String): Long

    /**
     * 根据角色查询资源
     */

    fun listMatchesRole(roles: Collection<String>): List<String>
}